src/Integration/Suppression/ChecksumCalculator.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+using System;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace SonarLint.VisualStudio.Integration.Suppression
+{
+    /// <summary>
+    /// Calculates the checksum for an issue using the same method as the SonarQube server
+    /// </summary>
+    /// <remarks>
+    /// For the corresponding code in SL IntelliJ see src\main\java\org\sonarlint\intellij\issue\LiveIssue.java::checksum
+    /// </remarks>
+    internal static class ChecksumCalculator
+    {
+        public static string Calculate(string text)
```

Comment:
The Java code is [here](https://github.com/SonarSource/sonarlint-intellij/blob/master/src/main/java/org/sonarlint/intellij/issue/LiveIssue.java#L92) and looks like this: 

`
private static int checksum(String content) {
    return Hex.encodeHexString(MD5_DIGEST.digest(content.replaceAll("[\\s]", "").getBytes(UTF_8))).hashCode();
  }
`

Note: the Java version also calls String.HashCode at the end. This isn't actually part of the checksum algorithm - they are calculating the checksum, then hashing it (not sure exactly why - I'll follow up with them and find out).